### ISSUE TITLE
Keyboard Shortcuts Fail with Hebrew Input – Need Universal Key Handling for Multilingual Support

### ISSUE THREAD

**Issue by minchaminder** [chucem/IveltChromeExtension#12]
**Description**:  
Currently, the way keyboard shortcuts are implemented for navigating and interacting with the site doesn't work when the keyboard is set to Hebrew input. The issue arises because the script relies on specific key strings (e.g., `"KeyA"`, `"KeyV"`) and `keydown` events tied to the English keyboard layout. When users switch to a Hebrew keyboard, these shortcuts no longer function as expected since the key codes and characters differ between languages.

**Explanation**:
Many users of the site often have their keyboard set to Hebrew while browsing and interacting with the site. Since Hebrew is the primary language for these users, it's common for them to input text, navigate, and perform actions with their keyboard set to Hebrew. However, the current keyboard shortcuts are designed to work only with English keyboard input, which means that when users have their keyboard set to Hebrew, these shortcuts fail to work.

This presents a significant usability issue, as users frequently need to switch back to English input just to use the site's navigation shortcuts, which interrupts their workflow. For users who may not be aware of this limitation, it can create confusion and frustration as the expected shortcuts simply don't respond.

Ensuring that keyboard shortcuts work seamlessly, regardless of the input language (Hebrew, English, or otherwise), is essential for maintaining a smooth, user-friendly experience for all users of the site.

### Problem:

- The current implementation checks for key codes that are specific to the English language, such as `"ArrowLeft"`, `"KeyA"`, and `"Enter"`.
- When the keyboard is set to Hebrew (or any other non-Latin input method), these keys no longer trigger the intended actions because the characters change in the input layer.

### Suggested Solution:

**Use Key Codes Instead of Key Strings**:
   Modify the script to rely on key codes (`e.keyCode`) or `e.code` instead of the actual key values like `"KeyA"`, `"KeyV"`. Key codes are more universal across languages and layouts. For example:
   - Instead of checking for `e.key == "KeyA"`, check for `e.code == "KeyA"`. The `.code` property gives you the physical key on the keyboard rather than the character associated with the key.

   ```javascript
   if (e.code == "KeyA" && e.altKey) {
       // Will trigger "active topics" regardless of input language
       window.location.href = 'https://www.ivelt.com/forum/search.php?search_id=active_topics';
   }
   ```

### Example of Revised Code:

```javascript
function checkKey(e) {
    if (e.code == "KeyA" && e.altKey) {
        window.location.href = 'https://www.ivelt.com/forum/search.php?search_id=active_topics';
    }

    if (e.code == "Enter" && e.ctrlKey && post) {
        sendPost(post);
    }

    if (e.code == "KeyV" && e.altKey && post) {
        previewPost();
    }

    if (e.code == "KeyN" && e.altKey) {
        toggleNotification();
    }

    if (e.code == "KeyM" && e.altKey) {
        nextNotification();
    }

    if (e.target.nodeName == "INPUT" || e.target.nodeName == "TEXTAREA" || e.target.isContentEditable) return;

    if (e.code == "ArrowLeft") {
        try {
            nextPage();
        } catch {
            console.log("attempted to go after last page");
        }
    } else if (e.code == "ArrowRight") {
        try {
            previousPage();
        } catch {
            console.log("attempted to go before first page");
        }
    } else if (e.code == "ArrowUp") {
        scrollTop();
    } else if (e.code == "ArrowDown") {
        scrollBottom();
    }
}
```

### Impact:
Without these changes, users who input Hebrew (or other non-Latin languages) cannot use the keyboard shortcuts effectively, causing usability issues on the site.

By updating the code to use key codes, the shortcut functionality will remain consistent regardless of the input language, improving the user experience for Hebrew and multilingual users.

**Comment by chucem** [comment]
what shortcut is not working for you? by me it works when my keyboard is on Hebrew

**Comment by minchaminder** [comment]
> what shortcut is not working for you? by me it works when my keyboard is on Hebrew

Right alt when in Hebrew mode

**Comment by chucem** [comment]
It has with some windows keyboards, not because of Hebrew.
the shortcut is `AltGraph` and need to be checked with `(e.altKey || e.getModifierState('AltGraph'))`

see https://jsfiddle.net/zcah60gq/4/

will fix in next version

**Comment by chucem** [comment]
fixed in v2.2.4
